imap_processing/cdf/imap_cdf_manager.py

```diff
@@ -12,9 +12,13 @@
 class ImapCdfAttributes(CdfAttributeManager):
     """Contains IMAP specific tools and settings for CDF management."""
 
-    def __init__(self):
+    def __init__(self, source_dir_input=None):
         """Set the path to the config directory."""
-        super().__init__(Path(__file__).parent / "config")
+        if source_dir_input is None:
+            super().__init__(Path(__file__).parent / "config")
+        else:
+            super().__init__(Path(__file__).parent / "config")
```

Comment:
This has actually been causing me a lot of weird struggles with testing this class. In CdfAttributeManager , the default schema and IMAPS default variables are in the "config" file path and not guaranteed in the  source_dir_input file path. I can't get the tests to run correctly without loading in that default info first, and then switching the file path... Thoughts?

Reply:
In those tests, you can pass in the "config" path (or pass in nothing and use the default) and then override it. This way it will still get the default schemas and global variables from "config" but you can also access your test files in the other file path.

Reply:
Hahaha, happy to say I just got it. Thank you so much! I see now.